aea/cli/issue_certificates.py

```diff
@@ -0,0 +1,103 @@
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Implementation of the 'aea issue_certificates' subcommand."""
+from pathlib import Path
+from typing import cast
+
+import click
+from click import ClickException
+
+from aea.cli.utils.config import load_item_config
+from aea.cli.utils.context import Context
+from aea.cli.utils.decorators import check_aea_project
+from aea.cli.utils.package_utils import get_package_path_unified
+from aea.configurations.base import ConnectionConfig, PublicId
+from aea.configurations.constants import CONNECTION
+from aea.crypto.registries import crypto_registry
+from aea.exceptions import enforce
+from aea.helpers.base import CertRequest
+
+
+@click.command()
+@click.pass_context
+@check_aea_project
+def issue_certificates(click_context):
+    """Issue certificates for connections that require them."""
+    ctx = cast(Context, click_context.obj)
+
+    for connection_id in ctx.agent_config.connections:
+        _process_connection(ctx, connection_id)
+
+    click.echo("Done!")
+
+
+def _process_certificate(ctx: Context, cert_request: CertRequest):
+    """Process a single certificate request."""
+    click.echo(f"Issuing certificate '{cert_request.identifier}'...")
+    ledger_id = cert_request.ledger_id
+    output_path = cert_request.path
+    if cert_request.key_identifier is not None:
+        key_identifier = cert_request.key_identifier
+        connection_private_key_path = ctx.agent_config.connection_private_key_paths.read(
+            key_identifier
+        )
+        if connection_private_key_path is None:
+            raise ClickException(
+                f"Cannot find connection private key with id '{key_identifier}'"
+            )
+        connection_crypto = crypto_registry.make(
+            key_identifier, private_key_path=connection_private_key_path
+        )
+        public_key = connection_crypto.public_key
+    else:
+        public_key = cast(str, cert_request.public_key)
+        enforce(
+            public_key is not None,
+            "Internal error - one of key_identifier or public_key must be not None.",
+        )
+    public_key_bytes = public_key.encode("ascii")
+    identifier = cert_request.identifier.encode("ascii")
+    not_before = cert_request.not_before_string.encode("ascii")
+    not_after = cert_request.not_after_string.encode("ascii")
+    crytpo_private_key_path = ctx.agent_config.connection_private_key_paths.read(
+        ledger_id
+    )
```

Comment:
No, for signing you use the non connection key. That way we establish the link

Reply:
yes, that was a typo, sorry